From f4166d683aeed6b5e17d8715ca676add112cb2e1 Mon Sep 17 00:00:00 2001
From: Kevin Brace <kevinbrace@gmx.com>
Date: Fri, 8 Jul 2016 14:43:31 -0500
Subject: [PATCH 12/21] Rewriting iga1_crtc_restore

Rewrote iga1_crtc_restore callback function, in order to improve
ACPI S3 State resume reliability.

Signed-off-by: Kevin Brace <kevinbrace@gmx.com>
---
 src/via_display.c | 16 ++++++----------
 1 file changed, 6 insertions(+), 10 deletions(-)

diff --git a/src/via_display.c b/src/via_display.c
index b52554a..7738516 100644
--- a/src/via_display.c
+++ b/src/via_display.c
@@ -2832,27 +2832,23 @@ iga1_crtc_restore(xf86CrtcPtr crtc)
     DEBUG(xf86DrvMsg(pScrn->scrnIndex, X_INFO,
                         "Entered iga1_crtc_restore.\n"));
 
-    vgaHWProtect(pScrn, TRUE);
-
-    /* Restore the standard VGA registers. */
-    if (xf86IsPrimaryPci(pVia->PciInfo)) {
-        vgaHWRestore(pScrn, &hwp->SavedReg, VGA_SR_ALL);
-    } else {
-        vgaHWRestore(pScrn, &hwp->SavedReg, VGA_SR_MODE);
-    }
-
     /* Gamma must be disabled before restoring palette. */
     ViaGammaDisable(pScrn);
 
+    vgaHWRestore(pScrn, &hwp->SavedReg, VGA_SR_ALL);
     viaIGA1Restore(pScrn);
 
     ViaDisablePrimaryFIFO(pScrn);
 
-    vgaHWProtect(pScrn, FALSE);
     vgaHWLock(hwp);
+    vgaHWProtect(pScrn, TRUE);
 
     DEBUG(xf86DrvMsg(pScrn->scrnIndex, X_INFO,
                         "Exiting iga1_crtc_restore.\n"));
+
+
+
+
 }
 
 static Bool
-- 
1.9.1

